src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -240,24 +303,170 @@ private String generateFrameDescription(WindowAggregationConfig config) {
-                             isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
-                             isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
+          isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
+          isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
       case RANGE:
         return String.format("over a windows frame consisting of %s preceding and %s following range.",
-                             isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
-                             isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
+          isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
+          isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
     }
 
     return "";
   }
 
   @Override
-  public JavaRDD<StructuredRecord> transform(
-    SparkExecutionPluginContext sparkExecutionPluginContext, JavaRDD<StructuredRecord> javaRDD) throws Exception {
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext sparkExecutionPluginContext,
+     JavaRDD<StructuredRecord> javaRDD) throws Exception {
     Schema inputSchema = sparkExecutionPluginContext.getInputSchema();
     if (inputSchema == null) {
       throw new Exception("Input schema is null. Input schema can not be null at this stage");
     }
     return WindowsAggregationUtil.transform(sparkExecutionPluginContext, javaRDD, config, inputSchema, outputSchema);
   }
+
+  @Override
+  public boolean canUseEngine(Engine engine) {
+    Optional<ExpressionFactory<String>> expressionFactory = engine.
+      getExpressionFactory(StringExpressionFactoryType.SQL);
+    return expressionFactory.isPresent();
+  }
+
+  //Generate window aggregation definition by setting values
+  @Nullable
+  public WindowAggregationDefinition generateAggregationDefinition(RelationalTranformContext ctx, Relation relation,
+    FailureCollector failureCollector, ExpressionFactory<String> expressionFactory) {
+    WindowAggregationConfig.WindowFrameType windowFrameType = config.getWindowFrameType();
+    WindowAggregationDefinition.WindowFrameType windowFrameTypeValue = null;
+    boolean unboundedPreceding = false;
+    boolean unboundedFollowing = false;
+    String preceding = null;
+    String following = null;
+    
+    if (windowFrameType != WindowAggregationConfig.WindowFrameType.NONE) {
+      unboundedPreceding = config.isFrameDefinitionUnboundedPreceding();
+      unboundedFollowing = config.isFrameDefinitionUnboundedFollowing();
+      preceding = String.valueOf(config.getFrameDefinitionPrecedingBound());
+      following = String.valueOf(config.getFrameDefinitionFollowingBound());
+      if (windowFrameType == WindowAggregationConfig.WindowFrameType.ROW) {
```

Comment:
you can implement this logic in the Enum. See https://github.com/data-integrations/wrangler/blob/3712f46455fd8389dc75062f420e4a704c286198/wrangler-service/src/main/java/io/cdap/wrangler/SamplingMethod.java as an example

Reply:
This logic also has the same validation.Do we have to implement additionally?

Reply:
Why do we need to implement in addition to existing functionality?

Reply:
It allows us to contain this logic to the enum, which makes the code easier to read and easier to test. You simply do Enum.valueOf(string).